### PR TITLE
helm/prometheus: don't limit selectors to MatchLabels

### PR DESCRIPTION
**Pull request by duboisph**
This updates the prometheus Helm chart so `MatchLabels` isn't hard
coded in the serviceMonitor- and ruleSectors. You'll have to specify
how to match via values.yaml.

Example usage:
```yaml
serviceMonitorsSelector:
  matchExpressions:
  - key: k8s-app
    operator: Exists

serviceMonitors:
  - name: "alertmanager"
    serviceMonitorSelectorLabels:
      k8s-app: alertmanager
    selector:
      matchLabels:
        alertmanager: main
    namespaceSelector:
      matchNames:
        - monitoring
    endpoints:
      - port: http
        interval: 30s
```

Also fixes/removes the `.Values.serviceMonitorSelector` typo in
the `servicemonitors.yaml` template.

This is a breaking change.

### CODE REVIEW

File: helm/prometheus/templates/servicemonitors.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ items:
         chart: {{ $chartName }}-{{ $chartVersion }}
         heritage: {{ $releaseService }}
         release: {{ $releaseName }}
-{{ toYaml .Values.serviceMonitorSelector | indent 8 }}
+{{ toYaml .labels | indent 8 }}
```
Comment (SamClinckspoor) [inline helm/prometheus/templates/servicemonitors.yaml]: why this name change? `labels` doesn't make it very apparent that this is whats used for selection

Comment (ringods) [inline helm/prometheus/templates/servicemonitors.yaml]: This `labels` value is not documented in the README.md. Better do this or the PR upstream will not be accepted.

Comment (SamClinckspoor) [inline helm/prometheus/templates/servicemonitors.yaml]: is it needed at all? the other resources have no provision to set custom labels

Comment (duboisph) [inline helm/prometheus/templates/servicemonitors.yaml]: Have you looked at the documentation? All sub values are not documented now :(

Comment (duboisph) [inline helm/prometheus/templates/servicemonitors.yaml]: @ringods ping ^

Comment (ringods) [inline helm/prometheus/templates/servicemonitors.yaml]: OK

Comment (SamClinckspoor) [inline helm/prometheus/templates/servicemonitors.yaml]: is it needed at all? the other resources have no provision to set custom labels

Comment (ringods) [inline helm/prometheus/templates/servicemonitors.yaml]: Yes, if you want to use a ServiceMonitor to monitor customer applications, they only need to attach a custom label to their apps and specify it here for the `prometheus-operator` to set up the correct monitoring.

Comment (SamClinckspoor) [inline helm/prometheus/templates/servicemonitors.yaml]: then it should be renamed, labels is too ambiguous

**Pull request merged**